FileShareService.DesktopClient.Core/AuthProvider.cs

```diff
@@ -17,10 +18,10 @@ public interface IAuthProvider : INotifyPropertyChanged
         DateTimeOffset? CurrentAccessTokenExpiry { get; }
         Task<string?> Login();
 
-        IEnumerable<string> Roles { get; }
+        IEnumerable<string> Roles { get; }       
     }
 
-    public class AuthProvider : BindableBase, IAuthProvider
+    public class AuthProvider : BindableBase, IAuthProvider, IAuthTokenProvider
```

Comment:
It looks to me that the app does not check for access-token expiry. I would prefer to see the `IsLoggedIn` property use `publicClientApplication.AcquireTokenSilent()` to determine if the access-token is still valid (and as a bonus side-effect `AcquireTokenSilent()` will automatically refresh the access-token if needed using a refresh-token).
This could require changing `publicClientApplication` to an instance variable and rebuilding it whenever the user switches environment.